tartufo/commands/pre_commit.py

```diff
@@ -1,23 +1,25 @@
 import pathlib
-from typing import List, Tuple
+from typing import Optional, Tuple
 
 import click
 
 from tartufo import types, util
-from tartufo.scanner import GitPreCommitScanner, Issue
+from tartufo.scanner import GitPreCommitScanner
 
 
 @click.command("pre-commit")
 @click.pass_obj
 @click.pass_context
-def main(ctx: click.Context, options: types.GlobalOptions) -> Tuple[str, List[Issue]]:
+def main(
+    ctx: click.Context, options: types.GlobalOptions
+) -> Tuple[str, Optional[GitPreCommitScanner]]:
     """Scan staged changes in a pre-commit hook."""
     # Assume that the current working directory is the appropriate git repo
     repo_path = pathlib.Path.cwd()
-    issues: List[Issue] = []
+    scanner = None
     try:
         scanner = GitPreCommitScanner(options, str(repo_path))
-        issues = scanner.scan()
+        scanner.scan()
     except types.ScanException as exc:
         util.fail(str(exc), ctx)
-    return (str(repo_path), issues)
+    return (str(repo_path), scanner)
```

Comment:
I really like this new structure for the commands. I think it reads really well. Thanks for making this happen!